syntax = "proto3";

package character;

service CharactersService {
    rpc Get(CharacterRequest) returns (CharacterResponse);
}

message CharacterRequest {
    CharacterType character_type = 1;
}

message CharacterResponse {
    Character character = 1;
}

enum CharacterType {
    UNKNOWN_CHARACTER = 0;
    HUMAN = 1;
    BEAST = 2;
}

message Character {
    CharacterType character_type = 1;
    int32 health = 2;
    int32 strength = 3;
    int32 defence = 4;
    int32 speed = 5;
    int32 chance = 6;
    repeated Skills skills = 7;
    string name = 8;
}

enum SkillType {
    UNKNOWN_SKILL = 0;
    DEFENCE = 1;
    ATTACK = 2;
}

message Skills {
    string name = 1; // skill name
    int32 chance = 2; // chance for your skill to perform
    SkillType skill_type = 3; // based on skill type different attribute is affected
    float power = 4; // increasing stat
}
